### PR TITLE
Create switch heading and use it some places

### PR DESCRIPTION
**Pull request by aeolianeth**
## What does this PR do and why?

- fixes the rules form
- adds a SwitchHeading component, uses it for the Token tab, hopefully fixes Token tab fo good.
- resets Token tab values from the Funding tab where appropriate

I'm fairly sure this checks off https://github.com/jbx-protocol/juice-interface/issues/787, but pls note anything i've missed or that is sus.

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the
changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

File: src/components/v2/V2Create/forms/FundingForm/index.tsx [hunk 1]
```diff
@@ -153,6 +159,24 @@ export default function FundingForm({ onFinish }: { onFinish: VoidFunction }) {
       )
       dispatch(editingV2ProjectActions.setDuration(durationInSeconds ?? '0'))
 
+      // reset discount rate if duration is 0
+      if (!durationInSeconds || durationInSeconds === '0') {
```
Comment (aeolianeth) [inline src/components/v2/V2Create/forms/FundingForm/index.tsx:163]: if someone goes to token tab, then back to funding tab and makes changes, then we might need to reset some values. we do that below

File: src/components/v2/V2Create/forms/RulesForm/index.tsx [hunk 2]
```diff
@@ -20,52 +20,35 @@ import {
...
-
-  const onFormSaved = useCallback(
-    (fields: RulesFormFields) => {
-      dispatch(editingV2ProjectActions.setPausePay(fields.pausePay))
-      dispatch(editingV2ProjectActions.setAllowMinting(fields.allowMinting))
-      dispatch(editingV2ProjectActions.setBallot(ballotStrategy.address))
-      onFinish?.()
-    },
-    [dispatch, onFinish, ballotStrategy],
+  const [pausePay, setPausePay] = useState<boolean>(
```
Comment (aeolianeth) [inline src/components/v2/V2Create/forms/RulesForm/index.tsx:39]: Totally bail on using the `form`. It's a massive pain to get working, there's no need for it here anyway.

File: src/components/v2/V2Create/forms/TokenForm/index.tsx [hunk 3]
```diff
@@ -84,49 +127,25 @@ export default function TokenForm({ onFinish }: { onFinish: VoidFunction }) {
...
+  ])
 
   return (
-    <Form
-      form={tokenForm}
-      initialValues={initialValues}
-      layout="vertical"
-      onFinish={onTokenFormSaved}
-    >
+    <Form layout="vertical" onFinish={onTokenFormSaved}>
```
Comment (aeolianeth) [inline src/components/v2/V2Create/forms/TokenForm/index.tsx:148]: Bail on `form` here too lol

File: src/components/v2/V2Create/tabs/FundingCycleTab/FundingCycleTabContent.tsx [hunk 4]
```diff
@@ -107,6 +107,7 @@ export default function FundingCycleTabContent({
           setFundingDrawerVisible(false)
         }}
         getContainer={false}
+        destroyOnClose
```
Comment (aeolianeth) [inline src/components/v2/V2Create/tabs/FundingCycleTab/FundingCycleTabContent.tsx:110]: this is needed to we force to load from Redux every time the drawer is opened.

File: src/hooks/v2/contractReader/ProjectReservedTokens.ts [hunk 5]
```diff
@@ -14,8 +14,8 @@ export default function useProjectReservedTokens({
     contract: V2ContractName.JBController,
     functionName: 'reservedTokenBalanceOf',
     args:
-      projectId && reservedRate
-        ? [BigNumber.from(projectId).toHexString(), reservedRate]
+      projectId && reservedRate && !projectId.eq(0)
```
Comment (aeolianeth) [inline src/hooks/v2/contractReader/ProjectReservedTokens.ts:17]: so this hook doesn't try to load every time (even ont eh create form)

File: src/components/v2/V2Create/forms/FundingForm/index.tsx (hunk 1 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Create/forms/FundingForm/index.tsx:163]: Yeah got it

File: src/components/v2/V2Create/forms/RulesForm/index.tsx (hunk 2 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Create/forms/RulesForm/index.tsx:39]: Yeah lol don't blame you I've been tempted. Good call here for sure

File: src/components/v2/V2Create/tabs/FundingCycleTab/FundingCycleTabContent.tsx (hunk 4 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Create/tabs/FundingCycleTab/FundingCycleTabContent.tsx:110]: Yep

**Comment by johnnyd-eth** [comment]
Nice @aeolianeth this looks rock-solid in the UI, I couldn't find any  Only question I have:

- Will put this to everyone else (I'm still undecided as to what I prefer), but wondering if enabling the switch should set the field to a value (e.g. discount rate to 10% rather than staying at 0%). It feels weird to 'enable' a field and have it do nothing

**Comment by johnnyd-eth** [comment]
Also, I removed the close #787 because I left a few more comments on it outside the description which I'll work on today

**Comment by johnnyd-eth** [comment]
But overall this is great! Approving and happy if we do the setting value thing in a follow-up if we end up deciding to do that

**Review by johnnyd-eth — APPROVED** [review]

**Pull request merged**